src/zenml/zen_server/routers/stack_deployment_endpoints.py

```diff
@@ -0,0 +1,135 @@
+from zenml.enums import StackDeploymentProvider
+from zenml.models import DeployedStack, StackDeploymentInfo
+from zenml.stack_deployments.utils import get_stack_deployment_class
+from zenml.zen_server.auth import AuthContext, authorize
+from zenml.zen_server.exceptions import error_response
+from zenml.zen_server.utils import (
+    handle_exceptions,
+)
+
+router = APIRouter(
+    prefix=API + VERSION_1 + STACK_DEPLOYMENT,
+    tags=["stacks"],
+    responses={401: error_response, 403: error_response},
+)
+
+
+@router.get(
+    INFO,
+    response_model=StackDeploymentInfo,
+)
+@handle_exceptions
+def get_stack_deployment_info(
+    provider: StackDeploymentProvider,
+    _: AuthContext = Security(authorize),
+) -> StackDeploymentInfo:
+    """Get information about a stack deployment provider.
+
+    Args:
+        provider: The stack deployment provider.
+
+    Returns:
+        Information about the stack deployment provider.
+    """
+    stack_deployment_class = get_stack_deployment_class(provider)
+    return StackDeploymentInfo(
+        provider=provider,
+        description=stack_deployment_class.description(),
+        instructions=stack_deployment_class.instructions(),
+        post_deploy_instructions=stack_deployment_class.post_deploy_instructions(),
+        permissions=stack_deployment_class.permissions(),
+        locations=stack_deployment_class.locations(),
+    )
+
+
+@router.get(
+    URL,
+)
+@handle_exceptions
+def get_stack_deployment_url(
+    request: Request,
+    provider: StackDeploymentProvider,
+    stack_name: str,
+    location: Optional[str] = None,
+    auth_context: AuthContext = Security(authorize),
+) -> Tuple[str, str]:
+    """Return the URL to deploy the ZenML stack to the specified cloud provider.
+
+    Args:
+        request: The FastAPI request object.
+        provider: The stack deployment provider.
+        stack_name: The name of the stack.
+        location: The location where the stack should be deployed.
+        auth_context: The authentication context.
+
+    Returns:
+        The URL to deploy the ZenML stack to the specified cloud provider
+        and a text description of the URL.
+    """
+    stack_deployment_class = get_stack_deployment_class(provider)
+    # Get the base server URL used to call this FastAPI endpoint
+    url = request.url.replace(path="").replace(query="")
+    # Use HTTPS for the URL
+    url = url.replace(scheme="https")
+
+    token = auth_context.access_token
+    assert token is not None
+
+    # A new API token is generated for the stack deployment
+    expires = datetime.datetime.utcnow() + datetime.timedelta(minutes=60)
+    api_token = token.encode(expires=expires)
+
+    return stack_deployment_class(
+        stack_name=stack_name, location=location
+    ).deploy_url(zenml_server_url=str(url), zenml_server_api_token=api_token)
+
+
+@router.get(
+    STACK,
+    response_model=Optional[DeployedStack],
```

Comment:
There is no reason for having this specified here if it matches the return type, on the contrary we already had issues because of it. 
```suggestion
```